lib/typescript-fast-incremental-preprocessor.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-env node */
+const fs = require('fs');
+const path = require('path');
+const tmp = require('tmp')
+const child_process = require('child_process');
+const funnel = require('broccoli-funnel');
+const mergeTrees = require('broccoli-merge-trees');
+const Plugin = require('broccoli-plugin');
+
+class FastTypescriptWatcher extends Plugin {
+  constructor(inputNode, options = {}) {
+    super([ inputNode ], {
+      annotation: options.annotation,
+      needsCache: false,
+      name: 'ember-cli-typescript-fast-watcher',
+    });
+
+    this.outputDir = options.outputDir;
+    this.ui = options.ui;
+  }
+
+  build() {
+    if (this.watcher) {
+      return;
+    }
+    const inputDir = this.inputPaths[0];
+    this.watcher = child_process.fork(
+      require.resolve('typescript/bin/tsc'), [
+        '--watch',
+        '--project', inputDir,
+        '--outDir', this.outputDir,
+        '--noEmit', 'false'
+      ], { silent: true }); // TODO: stdout is messy
+
+    // TODO: clean up process
+  }
+}
+
+function makeTempDir() {
+  // TODO: the directory won't get monitored if it's in tmp/
+  const dir = '.e-c-ts';
```

Comment:
This seems like a reasonable compromise to me; we'll just want to do two things if we run with it:

1. Document it.
2. Figure out how to automatically update `.gitignore` on installing the app, so that this directory is always excluded.